extension.js

```diff
@@ -1,11 +1,47 @@
 const cuMarkdown = require('markdown-it-church-slavonic');
+const vscode = require('vscode');
+var markdownit = require('markdown-it');
```

Comment:
`vscode` guidelines suggest not to use a separate copy of markdown engine. Lets remove it and just use renderes from `cuMarkdown` explicitly.

If you need new entry points in `cuMarkdown` for that, please let me know.

Reply:
```suggestion
'use strict';
var renderer = require('./renderer');

module.exports = {
  church_slavonic_plugin : function (md, options) {
    md.set({ xhtmlOut: true });  
    if (options && options.renderer) {
      switch (options.renderer) {
        case 'xml':
          md.renderer = new renderer.XMLRenderer();
          break;
        case 'latex':
          md.renderer = new renderer.LaTeXRenderer();
          break;
        default:
          throw new Error('unknow renderer: ' + options.renderer);
      }
    }
    md.use(
      require('./redbukva')
    ).use(
      require('./verse')
    ).use(
      require('./pagebreak')
    );  
  },
  renderXML : function renderXML(mdString) {
    var _renderXML = new renderer.XMLRenderer();
    return _renderXML.render(mdString);
  }
};
```

does the change above in the `markdown-it-church-Slavonic` make sense?

Reply:
That wont work, because `Renderer.render` signature is different (it takes parsed tokens, not string). And since we are constrained by using VSCode markdown instance, renderXML function need to take it as a parameter.

Maybe something like this (inspired by https://github.com/markdown-it/markdown-it/blob/master/lib/index.js#L540):
```
function renderXML(md, src, env) {
  env = env || {};

  return (new renderer.XMLRenderer()).render(md.parse(src, env), md.options, env);
};
```

Reply:
Exported functions also required `markdown-it` as a parameter. so I can exclude it from the `package.json` as it recommended by the design, however, I have to import see the [doc](https://github.com/markdown-it/markdown-it#usage-examples)

Reply:
```suggestion
var md = require('markdown-it')();
...
let xml = renderXml(md, data, { renderer: 'xml' });
```

Reply:
@GeorgePlotnikov markdown-it should not be required. Check out the original code that used markdown instance:
```
exports.activate = function activate() {
    return {
        extendMarkdownIt(md) {
            return md.use(cuMarkdown);
        }
    }
}
```
As demonstrated above, VSCode will pass the actual markdown instance to `extendMarkdown` extension point. You need to move all code that operates on markdown instance inside the `extendMarkdown`.

Reply:
As I see this usage is about extending the MD preview functionality, but not for creating new commands, see the [documentation](https://code.visualstudio.com/api/extension-guides/markdown-extension). The code I suggested works fine.